stats/lib/typings.d.ts

```diff
@@ -7,4 +7,5 @@ export interface Logger {
 export interface Filter {
   from: string;
   to: string;
+  address?: string;
```

Comment:
This filter is shared by all endpoints. Most endpoints don't support the `address` field.

Can we please create a different filter interface for endpoints that accept an optional `address` field. Also is `address` optional for those endpoints? Judging from your SQL query, this field is actually required.

```ts
export interface DateRangeFilter {
  from: string;
  to: string;
}

export interface AddressFilter {
  address: string;
}
```

Then you can use `DateRangeFilter & AddressFilter` for your new endpoint, and `DateRangeFilter` in all existing endpoints.

Reply:
Started the work here: #137

Reply:
Will follow up once landed 🙏

Reply:
@juliangruber This is not resolved yet. Please revert the change in this interface.

In my PR, I decided to describe the endpoint-specific filter properties inline in the "fetcher" definition.

```js
/**
 * @param {import('pg').Pool} pgPool
 * @param {import('./typings').DateRangeFilter & {nonZero?: 'true'}} filter
 */
export const fetchRetrievalSuccessRate = async (pgPool, filter) => {
 // ...
}
```